ouroboros-network/src/Ouroboros/Network/Protocol/Handshake/Version.hs

```diff
@@ -35,9 +36,22 @@ import Codec.SerialiseTerm
 
 -- | The set of versions supported by the local agent are described by a map
 -- keyed on the version identifier.
+--
+-- If one needs to combine multiple versions the simplest way is to use the
+-- 'Semigroup' instance.
+--
+-- >
+-- > fold $ (simpleSingletonVersions ...)
+-- >       :| [ (simpleSingletonVersions ...)
+-- >          , (simpleSingletonVersions ...)
+-- >          , ...
+-- >          ]
+-- >
+--
 newtype Versions vNum extra r = Versions
   { getVersions :: Map vNum (Sigma (Version extra r))
   }
+  deriving Semigroup via (Map vNum (Sigma (Version extra r)))
```

Comment:
Wouldn't `deriving Semigroup` also have worked? Or `deriving newtype Semigroup`?

Reply:
I am not sure.  But this is very explicit, so I like it over `deriving Semigroup`, `deriving newtype` - is equally good to me.